docs/plugins/npe2_getting_started.md

```diff
@@ -0,0 +1,286 @@
+(npe2-getting-started)=
+
+# npe2 getting started guide
+
+This guide will walk you through the steps to write an `npe2`-style napari
+plugin from scratch.
+
+At the end of this guide, you will have a working plugin that can be installed
+with `pip` and autodetected by napari.
+
+## Overview
+
+Plugins are just python packages. They include contributions that napari
+may use when performing tasks (like reading data), and a "manifest" file that
+tells napari where in the package to find these contributions.
+
+This guide covers `npe2`-style plugins. `npe2` plugins declare the
+functionality that they contribute in a file called the plugin manifest.
+
+Creating a new plugin involves the following steps:
+
+1. Configure a python package to use an npe2 manifest (e.g. by editing
+   `setup.cfg`):
+   - Create the _entry point group_.
+   - Make sure the manifest file is added to `package_data`.
+2. Create the plugin manifest file.
+
+This guide will walk you through the steps using a [cookiecutter][] template.
+cookiecutter is a command line utility that generates a python package for you
+after asking a few questions. In this case, the package contains all the
+boilerplate needed for documenting, testing and deploying your plugin.
+
+```{note}
+Minimally, a plugin could be just three files: `setup.cfg`, a python file, and
+the plugin manifest file.
+```
+
+## 1. Create the project using cookiecutter
+
+Install cookiecutter and use the template as follows:
+
+```bash
+pip install cookiecutter
+cookiecutter https://github.com/napari/cookiecutter-napari-plugin --checkout npe2
+```
+
+`cookiecutter` will ask you a series of questions
+about the functionality you want your plugin to provide. In this guide we'll
+focus on creating a reader that can read numpy ('\*.npy') files.
+
+```sh
+# questions asked when running cookiecutter:
+...
+plugin_name: my-npy-reader
+...
+include_reader_plugin [y]: y
+...
+```
+
+Some of the cookiecutter prompts are elided above. We named our plugin
+`my_npy_reader`. This is the name that will be used for the python package. It
+should conform to the [PEP8][] naming convention (short, all-lowercase names,
+using dashes instead of underscores).
+
+When the cookiecutter asked to include a reader plugin, we selected `y`, and
+in the next question we told cookiecutter that our reader should be invoked
+for files matching the `\*.npy` glob pattern.
+
+After answering all the prompts, `cookiecutter` will create a directory called
+`my_npy_reader` in the current directory that holds the generated files. It
+will look something like this:
+
+```
+my-npy-reader
```

Comment:
always felt folder structure was missing from our docs so I love this 🙏

Reply:
We're missing the .napari folder in here - may be worth including with a comment "napari hub specific customization"

Reply:
need advice. could you tell me what to add?

Reply:
added what we discussed.  thanks